src/functions/unreferencedReusableObject.ts

```diff
@@ -0,0 +1,28 @@
+import { IFunction } from '../types';
+
+export type OasReusableObjectsLocation = '#/definitions' | '#/components/schemas';
+
+export const unreferencedReusableObject: IFunction<{ reusableObjectsLocation: OasReusableObjectsLocation }> = (
+  data,
+  opts,
+  _paths,
+  otherValues,
+) => {
+  if (typeof data !== 'object' || data === null) return [];
+
+  const normalizedSource = otherValues.resolved.spec.source === undefined ? '' : otherValues.resolved.spec.source;
+
+  const defined = Object.keys(data).map(name => `${normalizedSource}${opts.reusableObjectsLocation}/${name}`);
+
+  const orphans = defined.filter(defPath => !otherValues.resolved.graph.hasNode(defPath));
+
+  return orphans.map(orphanPath => {
+    return {
+      message: 'Potential orphaned reusable object has been detected.',
+      path: orphanPath
+        .split('#')[1]
```

Comment:
This might break for invalid JSON pointers that does not start `#` and does not contain `#` anywhere.
I know it's unlikely and in general we shouldn't really account for that, but it's better to return nothing than to have a TypeError.

Reply:
I've added a test to cover this and ensure this won't regress later.